qiling/os/memory.py

```diff
@@ -63,6 +63,29 @@ def __read_string(self, addr: int) -> str:
     def __write_string(self, addr: int, s: str, encoding: str):
         self.write(addr, bytes(s, encoding) + b'\x00')
 
+    def __getitem__(self, key: Union[slice, int]) -> bytearray:
+        if isinstance(key, slice):
+            start = key.start
+            stop = key.stop
+            step = key.step
+
+            if step is not None and step != 1:
+                # step != 1 means we have to do copy, don't allow it.
+                raise IndexError("Only support slicing continous memory")
+
+            return self.ql.mem.read(start, stop - start)
+        elif isinstance(key, int):
+            return self.ql.mem.read(key, 1)
+        else:
+            raise KeyError("Wrong type of key")
+    
+    def __setitem__(self, key: Union[slice, int], value: Union[bytes, bytearray]):
```

Comment:
As far as I remember `mem.write` accepts only `bytes` and will fail on `bytearray` data,

Reply:
No, ctypes will marshal most bytes like objects for `uc_mem_write` though I didn't find some relevant docs.

Reply:
Okay... you are correct. `bytearray` is not acceptable.

Reply:
done